h2o-core/src/main/java/hex/ModelMetricsRegressionCoxPH.java

```diff
@@ -166,64 +174,343 @@ static Stats concordance(final Vec startVec, final Vec stopVec, final Vec eventV
     private static Stats concordanceStats(Vec.Reader startVec, Vec.Reader stopVec, Vec.Reader eventVec, List<Vec.Reader> strataVecs, Vec.Reader estimateVec, long length) {
       assert 0 <= length && length <= Integer.MAX_VALUE;
       
-      Stream<Integer> allIndexes = IntStream.range(0, (int) length).boxed();
+      Collection<List<Integer>> byStrata =
+        IntStream.range(0, (int) length)
+                 .filter(i -> !estimateVec.isNA(i) && !stopVec.isNA(i) && (null == startVec || !startVec.isNA(i)))
+                 .boxed()
+                 .collect(groupingBy(
+                      i -> strataVecs.stream()
+                              .map(v -> v.at8(i))
+                              .collect(toList())
+                 )).values();
+
+      return byStrata.stream()
+        .map(
+          indexes -> statsForAStrata(startVec, stopVec, eventVec, estimateVec, indexes)
```

Comment:
Why is this using the Vec API at all? If you know you have only a very limited number of indices - can it be collected to non-distributed structures and avoid all overhead with using the Vec API to access a single value?